packages/jsii/lib/compiler.ts

```diff
@@ -433,8 +433,10 @@ export class Compiler implements Emitter {
         continue;
       }
 
-      // eslint-disable-next-line @typescript-eslint/no-var-requires, @typescript-eslint/no-require-imports
-      const tsconfig = require(tsconfigFile);
+      const { config: tsconfig } = ts.readConfigFile(
+        tsconfigFile,
+        ts.sys.readFile,
+      );
```

Comment:
What's this for? Just better form to use `readConfigFile` instead of a json `require()`? I agree with that just wondering if its necessary for some reason.

Reply:
The thing here is that `tsconfig.json` files are not "standard" JSON - they allow comments and trailing commas. `require` and `readJson` will choke on that and fail, which is undesirable. This uses the same code that `tsc` itself uses, so we are "guaranteed" to succeed at loading that if it's a valid configuration file.